sklearn/decomposition/online_lda.py

```diff
@@ -0,0 +1,678 @@
+
+    verbose : int, optional (default=0)
+        Verbosity level.
+
+    random_state : int or RandomState instance or None, optional (default=None)
+        Pseudo Random Number generator seed control.
+
+
+    Attributes
+    ----------
+    components_ : array, [n_topics, n_features]
+        Topic word distribution. components_[i, j] represents word `j` in topic `i`.
+        In the literature, this is called `lambda`.
+
+    n_iter_ : int
+        Number of iteration.
+
+
+    References
+    ----------
+    [1] "Online Learning for Latent Dirichlet Allocation", Matthew D. Hoffman,
+        David M. Blei, Francis Bach, 2010
+
+    [2] "Stochastic Variational Inference", Matthew D. Hoffman, David M. Blei,
+        Chong Wang, John Paisley, 2013
+
+    [3] Matthew D. Hoffman's onlineldavb code. Link:
+        http://www.cs.princeton.edu/~mdhoffma/code/onlineldavb.tar
+
+    """
+
+    def __init__(self, n_topics=10, doc_topic_prior=None, topic_word_prior=None,
+                 learning_method='online', learning_decay=.7, learning_offset=10.,
+                 max_iter=10, batch_size=128, evaluate_every=-1, total_samples=1e6,
+                 perp_tol=1e-1, mean_change_tol=1e-3, max_doc_update_iter=100,
+                 n_jobs=1, verbose=0, random_state=None):
+        self.n_topics = n_topics
+        self.doc_topic_prior = doc_topic_prior
+        self.topic_word_prior = topic_word_prior
+        self.learning_method = learning_method
+        self.learning_decay = learning_decay
+        self.learning_offset = learning_offset
+        self.max_iter = max_iter
+        self.batch_size = batch_size
+        self.evaluate_every = evaluate_every
+        self.total_samples = total_samples
+        self.perp_tol = perp_tol
+        self.mean_change_tol = mean_change_tol
+        self.max_doc_update_iter = max_doc_update_iter
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+        self.random_state = random_state
+
+    def _check_params(self):
+        """
+        check model parameters
+        """
+        if self.n_topics <= 0:
+            err_msg = "Invalid 'n_topics' parameter: %r" % self.n_topics
+            raise ValueError(err_msg)
+
+        if self.total_samples <= 0:
+            err_msg = "Invalid 'total_samples' parameter: %r" % self.total_samples
+            raise ValueError(err_msg)
+
+        if self.learning_offset < 0:
+            err_msg = "Invalid 'learning_offset' parameter: %r" % self.learning_offset
+            raise ValueError(err_msg)
+
+        if self.learning_method not in ("batch", "online"):
+            err_msg = "Invalid 'learning_method' parameter: %r" % self.learning_method
+            raise ValueError(err_msg)
+
+    def _init_latent_vars(self, n_features):
+        """
+        Initialize latent variables.
+        """
+        self.rng_ = check_random_state(self.random_state)
+        self.n_iter_ = 1
+        self.n_features = n_features
```

Comment:
But actually there is no need to store this, we can always access `self.components_.shape[1]` if we need later instead.